lib/ansible/plugins/strategy/__init__.py

```diff
@@ -924,8 +925,9 @@ def _do_handler_run(self, handler, handler_name, iterator, play_context, notifie
                                 break
                 except AnsibleError as e:
                     for host in included_file._hosts:
-                        iterator.mark_host_failed(host)
-                        self._tqm._failed_hosts[host.name] = True
+                        if host.name not in self._tqm._failed_hosts:
```

Comment:
I wonder if we should prevent double failing hosts in one place in `iterator.mark_host_failed`.